sdk/runtime/runner.go

```diff
@@ -23,14 +23,18 @@ func Invoke(tc func(*RunEnv) error) {
 	// interval. When this "push_interval" is changed, you may want to change the scrape interval
 	// on the pushgateway
 	pushStopCh := make(chan struct{})
+	pushAllowedFailures := 3
 	go func() {
 		pushInterval := 5 * time.Second
 		for {
 			select {
 			case <-time.After(pushInterval):
 				err := runenv.MetricsPusher.Add()
 				if err != nil {
-					runenv.RecordMessage("error during periodic metric push: %w", err)
+					pushAllowedFailures -= 1
```

Comment:
We should match against the actual error to take this path. If the error is different (e.g. pushgateway is actually reachable but failing), we probably want to record it repeatedly.